### PR TITLE
Added fake auth flow integration tests

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

**Comment by kunagpal** [comment]
@czardoz Added 😈

File: test/integration/auth-methods/fake.test.js [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+
+            var err = testrun.request.firstCall.args[0],
+                request = testrun.request.firstCall.args[3];
+
+            expect(err).to.be.ok();
+            expect(request.url.toString()).to.eql('https://postman-echo.com/basic-auth');
+        });
+    });
+
+    describe('false result in pre helper', function () {
```
Comment (kunagpal) [inline test/integration/auth-methods/fake.test.js]: Done.

File: test/integration/auth-methods/fake.test.js [hunk 2]
```diff
@@ -0,0 +1,993 @@
...
+
+        it('must have sent the request once', function () {
+            expect(testrun.request.calledOnce).be.ok();
+
+            var err = testrun.request.firstCall.args[0],
+                request = testrun.request.firstCall.args[3];
+
+            expect(err).to.be(null);
+            expect(request.url.toString()).to.eql('https://postman-echo.com/basic-auth');
+        });
```
Comment (kunagpal) [inline test/integration/auth-methods/fake.test.js]: Added

File: test/integration/auth-methods/fake.test.js [hunk 3]
```diff
@@ -0,0 +1,993 @@
...
+    });
+
+    describe('error in post helper', function () {
+        var testrun,
+            fakeHandler = {
+                init: function (context, requester, done) { done(null); },
+                pre: function (context, requester, done) { done(null, true); },
+                post: function (context, requester, done) { done(new Error('Post err!')); },
```
Comment (shamasis) [inline test/integration/auth-methods/fake.test.js]: Tracked internally.

File: test/integration/auth-methods/fake.test.js [hunk 4]
```diff
@@ -0,0 +1,993 @@
...
+
+            var err = testrun.request.firstCall.args[0],
+                request = testrun.request.firstCall.args[3];
+
+            expect(err).to.be.ok();
+            expect(request.url.toString()).to.eql('https://postman-echo.com/basic-auth');
+        });
+    });
+
+    describe('error in pre, post helpers', function () {
```
Comment (shamasis) [inline test/integration/auth-methods/fake.test.js]: Tracked.

**Pull request merged**